### DISCUSSION TITLE
How to add values to surfaces and interpolate them in 3d?

### DISCUSSION THREAD

**Discussion by spaceswimmer** [gempy-project/gempy#925] [Q&A] [answered]
Hello! I know it was possible in earlier versions of GemPy to add values to surfaces, so they later would be interpolated in 3d. I wan't to do the same, but the .add_surface_values method doesn't exist anymore. Is there a way to add values to specific layers (for example acoustic waves speed) and after computing the model, getting them back as a 3d numpy array?
Thank you in advance

**Comment by javoha** [comment, marked as answer]
Hi,
I think you are correct that this functionality is not currently included in the base functionality of gempy v3. It should be rather straightforward to get what you want though: `geo_model.solutions.raw_arrays.lith_block` contains the resulting block model, meaning an ID value defining a structural element for each grid point. Just create a copy of this array an replace the values fpr each rock unit with a corresponding value (e.g acoustic speed wave),
Is that what you need/mean?
Cheers
Jan

**Reply by spaceswimmer** [reply]
Thanks a lot! That's what I was looking for!
